src/main/java/io/stargate/sgv2/jsonapi/service/processor/HybridFieldExpander.java

```diff
@@ -42,24 +43,45 @@ public static void expandHybridField(CommandContext context, Command command) {
   }
 
   // protected for testing purposes
-  protected static void expandHybridField(int docIndex, int docCount, JsonNode docNode) {
+  protected static void expandHybridField(
+      CommandContext context, int docIndex, int docCount, JsonNode docNode) {
     final JsonNode hybridField;
 
-    if ((docNode instanceof ObjectNode doc)
-        && (hybridField = doc.remove(DocumentConstants.Fields.HYBRID_FIELD)) != null) {
-      switch (hybridField) {
-        case NullNode ignored -> addLexicalAndVectorize(doc, hybridField, hybridField);
-        case TextNode ignored -> addLexicalAndVectorize(doc, hybridField, hybridField);
-        case ObjectNode ob -> addFromObject(doc, ob, docIndex, docCount);
-        default ->
-            throw ErrorCodeV1.HYBRID_FIELD_UNSUPPORTED_VALUE_TYPE.toApiException(
-                "expected String, Object or `null` but received %s (Document %d of %d)",
-                JsonUtil.nodeTypeAsString(hybridField), docIndex + 1, docCount);
+    if (docNode instanceof ObjectNode doc) {
+      // Check for $hybrid field
+      if ((hybridField = doc.remove(DocumentConstants.Fields.HYBRID_FIELD)) != null) {
+        switch (hybridField) {
+            // this is {"$hybrid" : null}
+          case NullNode ignored -> addLexicalAndVectorize(doc, hybridField, hybridField);
+          case TextNode ignored -> {
+            context.commandFeatures().addFeature(CommandFeature.HYBRID);
+            addLexicalAndVectorize(doc, hybridField, hybridField);
+          }
+          case ObjectNode ob -> addFromObject(context, doc, ob, docIndex, docCount);
+          default ->
+              throw ErrorCodeV1.HYBRID_FIELD_UNSUPPORTED_VALUE_TYPE.toApiException(
+                  "expected String, Object or `null` but received %s (Document %d of %d)",
+                  JsonUtil.nodeTypeAsString(hybridField), docIndex + 1, docCount);
+        }
+      } else {
+        // No $hybrid field, check other fields and add feature usage to CommandContext
+        if (doc.get(DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD) != null) {
```

Comment:
From discussions - move this to the Insert commands so they are tracking what they see. This class can just track if it sees the $hybrid feature